grin/src/Transformations/Optimising/CopyPropagation.hs

```diff
@@ -69,5 +69,8 @@ copyPropagation e = hylo folder builder (mempty, e) where
       | val == lpat
       , isConstant val
       -> rightExp
+    -- left unit law ; cleanup x <- pure y copies
```

Comment:
Write more detailed comment about the reason the dead variable removal is needed. It will not be obvious in the future.

Reply:
Done